wpilib/tests/test_relay.py

```diff
@@ -155,13 +170,60 @@ def test_relay_get(dir, value, fwd, rev, wpilib, relay_data):
     #hal.getRelayForward.assert_called_once_with(relay._port)
     #hal.getRelayReverse.assert_called_once_with(relay._port)
 
+
 def test_relay_get_freed(relay, hal):
     relay.free()
     #hal.reset_mock()
     with pytest.raises(ValueError):
         relay.get()
     #assert not hal.setRelayForward.called and not hal.setRelayReverse.called
 
+
+def test_relay_getChannel(relay):
+    assert relay.getChannel() == 2
+
+
+def test_relay_expiration(relay):
+    relay.setExpiration(111)
+    assert relay.getExpiration() == 111
+
+
+def test_relay_isAlive1(relay):
+    relay.setSafetyEnabled(False)
+    assert relay.isAlive()
+    relay.setSafetyEnabled(True)
+    assert not relay.isAlive()
+
+
+def test_relay_isAlive2(wpilib):
+    with patch('hal_impl.functions.hooks', new=SimHooks()) as hooks:
```

Comment:
Thanks for adding tests! However, it would be better if the sim hooks was a fixture in conftest.py instead of a standalone file:

```
@pytest.fixture
def sim_hooks(hal_data):
    with patch('hal_impl.functions.hooks', new=SimHooks()) as hooks:
        yield hooks
```

Reply:
with this syntax, when does hal_impl.functions.hooks get restored to its original value?

Reply:
If you use `@pytest.fixture(scope='function')` it will get restored after each test has ran.

Reply:
seems like my sim hooks breaks some tests in test_utility, does the fixture always get initialized, or only when it is requested?

Reply:
Fixtures only are initialized when requested.